### PR TITLE
[RCIAM-57] Users should be redirected to the sp they were trying to log in after registration

### PR DESCRIPTION
**Pull request by ioigoume**
- Redirect users to the service they requested after registration. Currently we are redirected in user's COManage profile
- Removed trailing spaces

**This functionality will be in COmanage 3.3 as a core feature.**

Update the schema of the database
alter table cm_co_petitions add column targetnew varchar(1024);

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Add `hidden` functionality to Enroll page. The Admin can enable the functionality by changing the value of `Hide Enrollment Flow` field to true, in the config page of an Enrollment flow. By default the value is false/empty and all the configured Enrolment Flows will be displayed in `People->Enroll` page.
 - Retrieve AuthenticatingAuthority and depict in CO Person's canvas/profile
 - IdP hinting for RCAUTH plugin
+- Redirect User to the SP after registration. Currently the User was stack in COmanage profile view and should go back and reselect the service.
```
Comment (NicolasLiampotis) [inline CHANGELOG.md]: the user was redirected to their COmanage profile

File: app/Controller/CoPetitionsController.php [hunk 2]
```diff
@@ -872,7 +886,10 @@ protected function dispatch($step, $id=null) {
         'action'     => $this->nextSteps[$step],
         $ptid
       );
-      
+      // RCIAM-57
+      if(isset($target_service_url) && $target_service_url != ""){
```
Comment (NicolasLiampotis) [inline app/Controller/CoPetitionsController.php:898]: Check if valid url

File: app/Controller/CoPetitionsController.php [hunk 3]
```diff
@@ -1369,6 +1386,17 @@ protected function execute_redirectOnFinalize($id) {
                                                               array('CoEnrollmentFlow.id' => $this->cachedEnrollmentFlowID));
     }
     
+    // RCIAM-57
+    // If a redirect on finalize is not available, check if a target new is
+    if(!$targetUrl || $targetUrl == "") {
```
Comment (NicolasLiampotis) [inline app/Controller/CoPetitionsController.php:1411]: Check if valid URL

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ioigoume) [reply CHANGELOG.md]: Done

File: app/Controller/CoPetitionsController.php (hunk 2 above)
Comment (ioigoume) [reply app/Controller/CoPetitionsController.php:898]: This should only happen at the entry point and not on every occasion.

Comment (ioigoume) [reply app/Controller/CoPetitionsController.php:898]: Done

File: app/Controller/CoPetitionsController.php (hunk 3 above)
Comment (ioigoume) [reply app/Controller/CoPetitionsController.php:1411]: Done. See comment from above

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**